### PR TITLE
using Cargo dep-info

### PR DESCRIPTION
**Pull request by declanqian**
#353

### CODE REVIEW

**Comment by declanqian** [comment]
We have to update `rust-toolchain` cause latest nightly fixed a issues regarding dep-info
[paths in `cargo:rerun-if-changed` are not properly resolved in depfiles](Relative paths in `cargo:rerun-if-changed` are not properly resolved in depfiles)

> As of (approximately) rust 1.52.0-nightly (a143517d4 2021-02-16), the const_in_array_repeat_expressions feature has been "removed due to causing promotable bugs." 

https://github.com/rust-osdev/x86_64/issues/233#issue-812497753

**Comment by declanqian** [comment]
I followed [`Alacritty`'s implementation](https://github.com/alacritty/alacritty/blob/master/alacritty/src/display/window.rs#L301).
As far as I can tell, PGTK Emacs doesn't do this explicitly, GTK does it for us by using the executable's name. So I did the same for WR Emacs.

**Comment by GTrunSec** [comment]
Can you do force-push with the latest upstream rev?

**Comment by GTrunSec** [comment]
looks good, just ignore this build error with nix

**Comment by declanqian** [comment]
Nice, I am pretty much finished with this task too.
Applied the same logic for `libremacs_lib`.

File: rust_src/crates/emacs/src/lib.rs [hunk 1]
```diff
@@ -10,7 +10,7 @@
 #![cfg_attr(test, allow(unused))]
 #![cfg_attr(feature = "strict", deny(warnings))]
 #![feature(concat_idents)]
-#![feature(const_fn)]
+// #![feature(const_fn)]
```
Comment (harryfei) [inline rust_src/crates/emacs/src/lib.rs:13]: Remove this line?

Comment (declanqian) [reply rust_src/crates/emacs/src/lib.rs:13]: Yeah, sure. Let me do that very quick.

Comment (declanqian) [reply rust_src/crates/emacs/src/lib.rs]: Removed that  line.

**Review by harryfei — APPROVED** [review]

**Comment by harryfei** [comment]
Is the macos build error normal?

**Comment by declanqian** [comment]
Just re-ran the macOS related jobs, passed this time.

**Pull request merged**